pyproject.toml

```diff
@@ -0,0 +1,2 @@
+[build-system]
+requires = ["setuptools", "wheel", "scikit-build", "cmake", "ninja"]
```

Comment:
This file defined the Python project settings, and the `[build-system]` section defines the build-system that `pip` will setup in an isolated environment. The `requires` table is the list of packages that `pip` will then install that are required to build the package: `wheel` to make wheels, `scikit-build` to run the CMake build, `cmake` so that `scikit-build` can run `cmake` and `ninja` for likely the same reason). `setuptools` might be included because `scikit-build`s entry point seems to be a patched version of `setuptools.setup`, `scikitbuild.setup`, called in `setup.py`.

Reply:
Thanks!

Reply:
In time, I understand [this file should be replacing `setup.py`](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/). I think we could actually move some more bits from `setup.py` to `pyproject.toml`, but we did not get around to it. [`scikit-build` shows the changing of build system through `setup.py`](https://scikit-build.readthedocs.io/en/latest/usage.html?highlight=toml), although that, IUC, could also be done in `pyproject.toml`. As always, documentation is not great, so let's leave the reduction or removal of `setup.py` (and `setup.cfg`) for a future attempt.

Reply:
Python packaging is a terrible mess. `pyproject` is at least a step in the right direction.